ingestion/src/metadata/utils/collections.py

```diff
@@ -0,0 +1,27 @@
+"""
+Uility classes for collections
+"""
+
+
+class CaseInsensitiveString(str):
+    """
+    A case-insensitive string. Useful for case-insensitive comparisons like SQL.
+    """
+
+    def __eq__(self, other):
+        return self.lower() == other.lower()
+
+    def __hash__(self):
+        return hash(self.lower())
```

Comment:
You should use `.casefold()` for caseless matching. https://docs.python.org/3/library/stdtypes.html#str.casefold.

e.g.
```python
>>> "fooßBar".lower()
'fooßbar'
>>> "fooßBar".casefold()
'foossbar'
```